libvast/builtins/operators/taste.cpp

```diff
@@ -0,0 +1,103 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/concept/parseable/numeric/integral.hpp>
+#include <vast/concept/parseable/vast/pipeline.hpp>
+#include <vast/error.hpp>
+#include <vast/logger.hpp>
+#include <vast/pipeline.hpp>
+#include <vast/plugin.hpp>
+#include <vast/table_slice.hpp>
+
+#include <arrow/type.h>
+
+namespace vast::plugins::taste {
+
+namespace {
+
+class taste_operator : public pipeline_operator {
+public:
+  explicit taste_operator(uint64_t limit) noexcept : limit_{limit} {
+    // nop
+  }
+
+  caf::error add(table_slice slice) override {
+    // Determine the number of remaining events for this schema.
+    auto remaining_it = remaining_.find(slice.schema());
+    if (remaining_it == remaining_.end()) {
+      remaining_it
+        = remaining_.emplace_hint(remaining_.end(), slice.schema(), limit_);
+    }
+    auto& remaining = remaining_it->second;
```

Comment:
Why not simply `remaining_[slice.schema()]`?

Reply:
That would default-construct the value to `0` if the schema hasn't been seen yet, while this is initializing the value to `limit_`.

Reply:
I've expanded the comment directly above to make this clearer.